src/mame/ussr/1801vp014.cpp

```diff
@@ -0,0 +1,324 @@
+// license:BSD-3-Clause
+// copyright-holders:Sergey Svishchev
+/***************************************************************************
+
+    K1801VP1-014 gate array (keyboard controller for the BK) and MS7008
+    keyboard emulation.
+
+    https://github.com/1801BM1/k1801/blob/master/014/
+    http://www.bk001x.ru/For_Forum/Files/Pereferia/Bkshka_MC7008_shema_klava_ext.pdf
+    https://retrocomputing.stackexchange.com/questions/27062/
+
+    "Stop" button is handled by a dedicated circuit.
+
+***************************************************************************/
+
+#include "emu.h"
+#include "1801vp014.h"
+
+#include "machine/keyboard.ipp"
+
+
+//**************************************************************************
+//  GLOBAL VARIABLES
+//**************************************************************************
+
+// device type definition
+DEFINE_DEVICE_TYPE(K1801VP014, k1801vp014_device, "1801vp1-014", "1801VP1-014")
+
+
+//**************************************************************************
+//  MACROS / CONSTANTS
+//**************************************************************************
+
+//**************************************************************************
+//  LIVE DEVICE
+//**************************************************************************
+
+INPUT_PORTS_START(ms7008)
```

Comment:
It would be nice to have Russian key cap labels for the letter keys as well.  Also, this should be `static INPUT_PORTS_START(...)`.